dart/lib/src/http_client/breadcrumb_client.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:http/http.dart';
+import '../protocol.dart';
+import '../hub.dart';
+import '../hub_adapter.dart';
+
+/// A [http](https://pub.dev/packages/http)-package compatible HTTP client
+/// which records requests as breadcrumbs.
+///
+/// Remarks:
+/// If this client is used as a wrapper, a call to close also closes the
+/// given client.
+///
+/// The `BreadcrumbClient` can be used as a standalone client like this:
+/// ```dart
+/// import 'package:sentry/sentry.dart';
+///
+/// var client = BreadcrumbClient();
+/// try {
+///  var uriResponse = await client.post('https://example.com/whatsit/create',
+///      body: {'name': 'doodle', 'color': 'blue'});
+///  print(await client.get(uriResponse.bodyFields['uri']));
+/// } finally {
+///  client.close();
+/// }
+/// ```
+///
+/// The `BreadcrumbClient` can also be used as a wrapper for your own
+/// HTTP [Client](https://pub.dev/documentation/http/latest/http/Client-class.html):
+/// ```dart
+/// import 'package:sentry/sentry.dart';
+/// import 'package:http/http.dart' as http;
+///
+/// final myClient = http.Client();
+///
+/// var client = BreadcrumbClient(client: myClient);
```

Comment:
this would be a breaking change, not sure if its worth it

Reply:
No not really, because the new SentryHttpClient combines both clients.